tests/Support/SupportLazyCollectionIsLazyTest.php

```diff
@@ -782,11 +782,31 @@ public function testOnlyIsLazy()
             $collection->only(5, 6, 7);
         });
 
-        $this->assertEnumerates(8, function ($collection) {
+        $this->assertEnumeratesOnce(function ($collection) {
             $collection->only(5, 6, 7)->all();
         });
     }
 
+    public function testOnlyDoesNotCloseGenerators()
+    {
+        $generator = function () {
+            yield 'foo' => 1;
+            yield 'bar' => 2;
+            yield 'baz' => 3;
+        };
+
+        $actual = LazyCollection::times(2, fn () => new LazyCollection(fn () => yield from $generator()))
+            ->map(fn (LazyCollection $nestedCollection) => $nestedCollection->only('foo', 'bar'))
+            ->toArray();
+
+        $expected = [
+            ['foo' => 1, 'bar' => 2],
+            ['foo' => 1, 'bar' => 2],
+        ];
+
+        $this->assertSame($expected, $actual);
+    }
```

Comment:
@cerbero90 This test seems to be passing even without the change above.

If I add back the `if...empty...break` block, this test still passes.

What am I missing?